packages/core/src/createConfig.ts

```diff
@@ -319,6 +319,14 @@ export function createConfig<
     store.setState((x) => {
       const connector = connectors.getState().find((x) => x.uid === data.uid)
       if (!connector) return x
+
+      if (connector.emitter.listenerCount('connect'))
+        connector.emitter.off('connect', change)
+      if (!connector.emitter.listenerCount('change'))
+        connector.emitter.on('change', change)
+      if (!connector.emitter.listenerCount('disconnect'))
+        connector.emitter.on('disconnect', disconnect)
```

Comment:
`'connect'` event should bind connector emitter similar to how the [`connect` action does](https://github.com/wevm/wagmi/blob/main/packages/core/src/actions/connect.ts#L61-L63)